internal/common/utils/misc.go

```diff
@@ -109,3 +109,15 @@ func LoadSchemas(cfg *viper.Viper, schemaNames []string) (schemas []*jsonschema.
 	}
 	return
 }
+
+type apiVersionCtxKeyType string
+
+const apiCtxKey apiVersionCtxKeyType = "api_version"
+
+func WithApiVersion(ctx context.Context, value string) context.Context {
+	return context.WithValue(ctx, apiCtxKey, value)
+}
+
+func GetApiVersion(ctx context.Context) string {
+	return ctx.Value(apiCtxKey).(string)
```

Comment:
Should this somehow be called from a codepath where WithApiVersion was not called previously, this conversion will panic and the app will crash. As the metrics are not vital it would be better to default to some value instead (e.g. "unknown")